src/contact.rs

```diff
@@ -189,6 +190,128 @@ pub async fn make_vcard(context: &Context, contacts: &[ContactId]) -> Result<Str
     Ok(contact_tools::make_vcard(&vcard_contacts))
 }
 
+/// Imports contacts from the given vCard.
+///
+/// Returns the ids of created/modified contacts in the order they appear in `vcard`.
+pub async fn import_vcard(context: &Context, vcard: &str) -> Result<Vec<ContactId>> {
+    let contacts = contact_tools::parse_vcard(vcard);
+    let mut contact_ids = Vec::with_capacity(contacts.len());
+    for c in &contacts {
+        let Ok(Some(id)) = import_vcard_contact(context, c)
+            .await
+            .with_context(|| format!("import_vcard_contact() failed for {}", c.addr))
+            .log_err(context)
+        else {
+            continue;
+        };
+        contact_ids.push(id);
+    }
+    Ok(contact_ids)
+}
+
+async fn import_vcard_contact(
+    context: &Context,
+    contact: &VcardContact,
+) -> Result<Option<ContactId>> {
+    let (id, modified) = match ContactAddress::new(&contact.addr) {
+        Err(e) => return Err(e).context("Invalid address"),
+        Ok(addr) => {
+            // Maybe we haven't yet send messages to this contact, but importing the vCard is also
+            // an explicit user action, so it's ok. We mustn't use `Origin::AddressBook` here
+            // because the vCard may be created not by us, also we want `contact.authname` to be
+            // saved as the authname and not a locally given name.
+            let origin = Origin::OutgoingTo;
```

Comment:
It's not only "some known contact introduced us", but also "the user intentionally imports the vCard", that's why it's `OutgoingTo` because sending a message is also an explicit user action, and also probably the user is going to communicate to the imported contact

Reply:
I changed this to `Origin::CreateChat`:
```
/// a chat was manually created for this user, but no message yet sent
```